skada/_subspace.py

```diff
@@ -366,3 +366,233 @@ def TransferComponentAnalysis(
+        M = np.block([[Mss, Mst], [Mst.T, Mtt]])
+        return M
+
+    def _get_kernel_matrix(self, X_source, X_target):
+        Kss = pairwise_kernels(X_source, metric=self.kernel)
+        Ktt = pairwise_kernels(X_target, metric=self.kernel)
+        Kst = pairwise_kernels(X_source, X_target, metric=self.kernel)
+        K = np.block([[Kss, Kst], [Kst.T, Ktt]])
+        return K
+
+    def frobenius_norm(self, M):
+        return np.sqrt(np.trace(M @ M.T))
+
+    def l21_norm(self, A):
+        norm = 0
+        for i in range(A.shape[0]):
+            norm += np.linalg.norm(A[i])
+        return norm
+
+    def fit(self, X, y=None, sample_domain=None, **kwargs):
+        """Fit adaptation parameters.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The source data.
+        y : array-like, shape (n_samples,)
+            The source labels.
+        sample_domain : array-like, shape (n_samples,)
+            The domain labels (same as sample_domain).
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X, sample_domain = check_X_domain(
+            X,
+            sample_domain,
+            allow_multi_source=True,
+            allow_multi_target=True,
+        )
+        X_source, X_target = source_target_split(X, sample_domain=sample_domain)
+
+        if self.n_components is None:
+            n_components = min(min(X_source.shape), min(X_target.shape))
+        else:
+            n_components = self.n_components
+        self.random_state_ = check_random_state(self.random_state)
+
+        n = X.shape[0]
+        ns = (sample_domain > 0).shape[0]
+        H = np.identity(n) - 1 / n * np.ones((n, n))
+        M = self._get_mmd_matrix(X_source.shape[0], X_target.shape[0], sample_domain)
+        self.X_source_ = X_source
+        self.X_target_ = X_target
+        K = self._get_kernel_matrix(X_source, X_target)
+        M = M / self.frobenius_norm(M)
+        G = np.identity(n)
+
+        self.A_ = np.identity(n)[:, n - n_components :]
+        for i in range(self.max_iter):
+            B = self.regularizer * G + K @ M @ K
+            C = K @ H @ K
+            phi, A = eig(B, C)
+            indices = np.argsort(np.abs(phi))
+            A = A[indices]
+            A = np.real(A[:, n - n_components :])
+            for j in range(n):
+                if sample_domain[j] < 0:
+                    G[j, j] = 1
+                else:
+                    a = A[j]
+                    if np.array_equal(a, np.zeros(a.shape)):
+                        G[j, j] = 0
+                    else:
+                        G[j, j] = 1 / (2 * np.linalg.norm(a))
+            self.A_ = A
+            l21 = self.l21_norm(A[:ns])
+            f = self.frobenius_norm(A[ns:])
+            c = np.trace(A.T @ K @ M @ K @ A) + self.regularizer * (l21 + f**2)
+            print(c)
+
+        return self
+
+
+def TJM(
```

Comment:
do we really need a function for this? this seems redundant with the estimator class